### PR TITLE
Add #[must_use] to functions annotated with __attribute__((warn_unused_result))

### PR DESCRIPTION
**Pull request by flowbish**
First pass at implementing #1450. Take a look at the tests to see the behavior difference on rustc 1.27.

r? @emilio

### CODE REVIEW

File: src/ir/function.rs [hunk 1]
```diff
@@ -387,6 +392,24 @@ impl FunctionSig {
             }
         };
 
+        let mut must_use = false;
```
Comment (emilio) [inline src/ir/function.rs]: Could you move all this block to a function in `clang.rs`, probably a method of `cursor`?

Something like the following maybe:

```rust
pub fn has_simple_attr(&self, attr: &str) -> bool;
```

Then this code could just be:

```rust
let must_use = cursor.has_simple_attribute("warn_unused_result");
```

WDYT?

File: src/ir/function.rs [hunk 2]
```diff
@@ -310,12 +313,14 @@ impl FunctionSig {
...
         arguments: Vec<(Option<String>, TypeId)>,
         is_variadic: bool,
+        must_use: bool,
         abi: Abi,
     ) -> Self {
         FunctionSig {
             return_type: return_type,
             argument_types: arguments,
             is_variadic: is_variadic,
+            must_use: must_use,
```
Comment (emilio) [inline src/ir/function.rs]: I know it's pre-existing, but if you want this can just be `must_use,` (same for the other members).

**Comment by flowbish** [comment]
Alright, pushed some changes to address your comments. Thanks for the quick review!

File: src/ir/function.rs [hunk 3]
```diff
@@ -308,14 +311,16 @@ impl FunctionSig {
...
     ) -> Self {
         FunctionSig {
-            return_type: return_type,
-            argument_types: arguments,
-            is_variadic: is_variadic,
+            return_type,
+            argument_types,
+            is_variadic,
+            must_use,
             abi: abi,
```
Comment (emilio) [inline src/ir/function.rs]: nit: Could've done this for `abi` as well, but no big deal in any case :)

**Comment by emilio** [comment]
Ah, forgot bors is still dead.

**Pull request merged**